### DISCUSSION TITLE
Continuously Updating a Variable/Value

### DISCUSSION THREAD

**Discussion by AddamOrigo** [OpenModelica/OpenModelica#10955] [Q&A]
Hello, everyone -

I'm looking to simulate the rise and fall of water levels in a tank, and have a variable tied to the initial fill level. It is set initially to 400 gallons. With each iteration of the simulation, it returns a positive or negative value which either adds or removes a certain amount of water. 

Ideally, each time the water variable is calculated, I would add it to the water level, allowing it to either increase or decrease with each iteration. Something like: 
`tank_water_level += water`

I'd like to plot this with time, but can't determine what syntax I need to use to make this work.

Thank you for your help.

**Comment by phannebohm** [comment]
I'm not entirely sure what you try to accomplish. If `water` describes the rate of change (i.e. derivative w.r.t. time) of `tank_water_level` then you can just write the equation
```modelica
der(tank_water_level) = water;
```
and let the simulation runtime do the integration. In case you are new to Modelica you may find [this book](https://mbe.modelica.university/) helpful.

**Comment by casella** [comment]
I would rather write something like
```
A*rho*der(tank_level) = mass_flow_rate;
```
or
```
A*der(tank_level) = volume_flow_rate;
```
which is physically meaningful. Modelica was born to describe continuous-time systems via differential-algebraic equations. The tool automatically figures out how to solve them.

BTW, you should always use SI units, which are guaranteed to always be consistent.
